lib/quandl/api_config.rb

```diff
@@ -4,29 +4,36 @@ class ApiConfig
     API_BASE_THREAD_KEY = 'quandl_api_base'
     API_VERSION_THREAD_KEY = 'quandl_api_version_key'
 
+    @api_key = nil
+    @api_base = nil
+    @api_version = nil
+
     class << self
       def api_key=(api_key)
+        @api_key = api_key if @api_key.nil?
         Thread.current[API_KEY_THREAD_KEY] = api_key
       end
 
       def api_key
-        Thread.current[API_KEY_THREAD_KEY]
+        Thread.current[API_KEY_THREAD_KEY] || @api_key
       end
 
       def api_base=(api_base)
+        @api_base = api_base if @api_base.nil?
         Thread.current[API_BASE_THREAD_KEY] = api_base
       end
 
       def api_base
-        Thread.current[API_BASE_THREAD_KEY] || 'https://www.quandl.com/api/v3'
+        Thread.current[API_BASE_THREAD_KEY] || @api_base || 'https://www.quandl.com/api/v3'
```

Comment:
maybe add a test to ensure that the default is still `https://www.quandl.com/api/v3` if nothing is set